Babylon/commands/azure/ad/app/create.py

```diff
@@ -1,45 +1,84 @@
     https://learn.microsoft.com/en-us/graph/api/application-post-applications
     """
+    check_ascii(name)
     route = "https://graph.microsoft.com/v1.0/applications"
-    env = Environment()
-    payload_template = env.working_dir.payload_path / "webapp/app_registration.json"
-    registration_file = registration_file or payload_template
+    registration_file = registration_file or env.working_dir.original_template_path / "webapp/app_registration.json"
     details = env.fill_template(registration_file, data={"app_name": name})
-    response = oauth_request(route, azure_token, type="POST", data=details)
-    if response is None:
-        return CommandResponse.fail()
-    output_data = response.json()
+
+    handler = polling2.poll(lambda: oauth_request(route, azure_token, type="POST", data=details),
+                            check_success=is_correct_response_app,
+                            step=1,
+                            timeout=60)
+
+    output_data = handler.json()
+
     # Service principal creation
     sp_route = "https://graph.microsoft.com/v1.0/servicePrincipals"
-    sp_response = oauth_request(sp_route, azure_token, type="POST", json={"appId": output_data["appId"]})
+
+    sp_response = polling2.poll(
+        lambda: oauth_request(sp_route, azure_token, type="POST", json={"appId": output_data['appId']}),
+        check_success=is_correct_response_serviceprincipal,
+        step=1,
+        timeout=60)
+    sp_response = sp_response.json()
     if sp_response is None:
         logger.error("Failed to create application service principal")
         return CommandResponse.fail()
-    output_data["servicePrincipalId"] = sp_response.json()["id"]
-    return CommandResponse.success(output_data, verbose=True)
+    if select:
+        r_id = ctx.parent.command.name
+        env.configuration.set_var(resource_id=r_id, var_name="app_id", var_value=sp_response["appId"])
+        logger.info(SUCCESS_CONFIG_UPDATED("app", "app_id"))
+        env.configuration.set_var(resource_id=r_id, var_name="name", var_value=sp_response["appDisplayName"])
+        logger.info(SUCCESS_CONFIG_UPDATED("app", "name"))
+        env.configuration.set_var(resource_id=r_id, var_name="principal_id", var_value=sp_response["id"])
+        logger.info(SUCCESS_CONFIG_UPDATED("app", "principal_id"))
+        env.configuration.set_var(resource_id=r_id, var_name="object_id", var_value=output_data["id"])
+        logger.info(SUCCESS_CONFIG_UPDATED("app", "object_id"))
+    return CommandResponse.success(sp_response, verbose=True)
+
+
+def is_correct_response_app(response):
+    if response is None:
+        return CommandResponse.fail()
+    output_data = response.json()
+    if "id" in output_data:
+        return True
+
+
+def is_correct_response_serviceprincipal(response):
+    if response is None:
+        return CommandResponse.fail()
+    output_data = response.json()
+    if "id" in output_data:
+        return True
```

Comment:
I think here we could go for a unique function, let's call it 'is_correct_response' which responsibility would be to return a boolean if the given condition is met